### PR TITLE
BAEL-6055 NonUniqueResultException in Spring Data JPA

### PR DESCRIPTION
**Pull request by Basler182**

### CODE REVIEW

File: persistence-modules/spring-data-jpa-query-3/src/test/java/com/baeldung/spring/data/jpa/query/nonuniqueresultexception/NonUniqueResultExceptionIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import java.text.SimpleDateFormat;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest(properties = "spring.sql.init.data-locations=classpath:import_entities.sql", showSql = false)
+public class NonUniqueResultExceptionIntegrationTest {
+
+    @Autowired
+    private ArticleRepository repository;
+
+    @Test(expected = IncorrectResultSizeDataAccessException.class)
```
[the remark this answers is not in the document]
Comment (Basler182) [reply persistence-modules/spring-data-jpa-query-3/src/test/java/com/baeldung/spring/data/jpa/query/nonuniqueresultexception/NonUniqueResultExceptionIntegrationTest.java]: @Bwvolleyball Uh I like the idea very much. I have adapted the code according to your suggestion.

File: persistence-modules/spring-data-jpa-query-3/src/test/java/com/baeldung/spring/data/jpa/query/nonuniqueresultexception/NonUniqueResultExceptionIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+@RunWith(SpringRunner.class)
+@DataJpaTest(properties = "spring.sql.init.data-locations=classpath:import_entities.sql", showSql = false)
+public class NonUniqueResultExceptionIntegrationTest {
+
+    @Autowired
+    private ArticleRepository repository;
+
+    @Test
+    public void givenImportedArticlesWhenFindByPublicationTimeBetweenThenNIncorrectResultSizeDataAccessExceptionThrown() {
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa-query-3/src/test/java/com/baeldung/spring/data/jpa/query/nonuniqueresultexception/NonUniqueResultExceptionIntegrationTest.java:25]:
```suggestion
    public void givenImportedArticles_whenFindByPublicationTimeBetween_thenIncorrectResultSizeDataAccessExceptionThrown() {
```

File: persistence-modules/spring-data-jpa-query-3/src/test/java/com/baeldung/spring/data/jpa/query/nonuniqueresultexception/NonUniqueResultExceptionIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        assertThatThrownBy(() -> repository.findByPublicationTimeBetween(new SimpleDateFormat("HH:mm").parse("15:15"), new SimpleDateFormat("HH:mm").parse("16:30")))
+                .isInstanceOf(IncorrectResultSizeDataAccessException.class)
+                .hasCauseInstanceOf(NonUniqueResultException.class);
+    }
+
+    @Test
+    public void givenImportedArticlesWhenfindAllByPublicationTimeBetweenThenSuccess() throws ParseException {
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa-query-3/src/test/java/com/baeldung/spring/data/jpa/query/nonuniqueresultexception/NonUniqueResultExceptionIntegrationTest.java:32]:
```suggestion
    public void givenImportedArticles_whenFindAllByPublicationTimeBetween_thenSuccess() throws ParseException {
```

**Review by KevinGilmore — COMMENTED** [review]
Let's use underscores as per guidelines to make the test method names more readable (and correct the typo in the first one)

**Pull request merged**